src/Catalyst.Core.Lib.Tests/UnitTests/Rpc/IO/Observers/PeerBlackListingRequestObserverTests.cs

```diff
@@ -0,0 +1,153 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Linq;
+using Catalyst.Abstractions.IO.Messaging.Dto;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.Network;
+using Catalyst.Core.Lib.P2P.Models;
+using Catalyst.Core.Lib.P2P.Repository;
+using Catalyst.Core.Modules.Rpc.Server.IO.Observers;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Wire;
+using Catalyst.Protocol.Rpc.Node;
+using Catalyst.TestUtils;
+using DotNetty.Transport.Channels;
+using FluentAssertions;
+using Microsoft.Reactive.Testing;
+using NSubstitute;
+using Serilog;
+using Xunit;
+
+namespace Catalyst.Core.Lib.Tests.UnitTests.Rpc.IO.Observers
+{
+    /// <summary>
+    /// Tests the peer black listing calls
+    /// </summary>
+    public sealed class PeerBlackListingRequestObserverTests
+    {
+        private readonly ILogger _logger;
+
+        private readonly IChannelHandlerContext _fakeContext;
+
+        private readonly TestScheduler _testScheduler;
+        private readonly PeerId _senderId;
+        private readonly IPeerRepository _peerRepository;
+
+        public PeerBlackListingRequestObserverTests()
+        {
+            _logger = Substitute.For<ILogger>();
+            _fakeContext = Substitute.For<IChannelHandlerContext>();
+            
+            var fakeChannel = Substitute.For<IChannel>();
+            _fakeContext.Channel.Returns(fakeChannel);
+            _fakeContext.Channel.RemoteAddress.Returns(EndpointBuilder.BuildNewEndPoint("192.0.0.1", 42042));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make sure using this hardcoded IP address '192.0.0.1' is safe here.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)